src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalescingAssignmentTests.cs

```diff
@@ -0,0 +1,1408 @@
+  IL_00e8:  ldsfld     ""System.Runtime.CompilerServices.CallSite<System.Func<System.Runtime.CompilerServices.CallSite, dynamic, int, dynamic, dynamic>> C.<>o__3.<>p__1""
+  IL_00ed:  ldloc.0
+  IL_00ee:  ldc.i4.0
+  IL_00ef:  ldc.i4.2
+  IL_00f0:  box        ""int""
+  IL_00f5:  callvirt   ""dynamic System.Func<System.Runtime.CompilerServices.CallSite, dynamic, int, dynamic, dynamic>.Invoke(System.Runtime.CompilerServices.CallSite, dynamic, int, dynamic)""
+  IL_00fa:  br.s       IL_00fd
+  IL_00fc:  ldloc.1
+  IL_00fd:  callvirt   ""int? System.Func<System.Runtime.CompilerServices.CallSite, dynamic, int?>.Invoke(System.Runtime.CompilerServices.CallSite, dynamic)""
+  IL_0102:  ret
+}
+");
+        }
+
+        [Fact]
+        public void DynamicRHS()
+        {
+            CompileAndVerify(@"
+using System;
+using System.Collections.Generic;
+public class C
+{
+    public static void Main()
+    {
+        int? d = null;
+        Console.WriteLine(d ??= GetDynamicClass().Property1);
+        Console.WriteLine(d ??= GetDynamicClass().Property2);
+    }
+
+    public static dynamic GetDynamicClass() => new DynamicClass();
+}
+public class DynamicClass
+{
+    public int Property1
+    {
+        get
+        {
+            Console.WriteLine(""In get_Property1"");
+            return 1;
+        }
+    }
+    public int Property2
+    {
+        get
+        {
+            Console.WriteLine(""In get_Property2"");
+            return 2;
+        }
+    }
+}
+", new[] { CSharpRef }, expectedOutput: @"
+In get_Property1
+1
+1
+");
+        }
+
+        [Fact]
+        public void UseBeforeAssignment()
+        {
+            var compilation = CreateCompilation(@"
+public class C
+{
+    public static void Main()
+    {
+        C c;
+        c ??= new C();
+
+        object x1;
+        object y1;
+        GetC(x1 = 1).Prop ??= (y1 = 2);
+        x1.ToString();
+        y1.ToString();
+
+        object x2;
+        object y2;
+        GetC(x2 = 1).Field ??= (y2 = 2);
+        x2.ToString();
+        y2.ToString();
+    }
+    static C GetC(object i) => null;
+    object Prop { get; set; }
+    object Field;
+}").VerifyDiagnostics(
+                // (7,9): error CS0165: Use of unassigned local variable 'c'
+                //         c ??= new C();
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "c").WithArguments("c").WithLocation(7, 9),
+                // (13,9): error CS0165: Use of unassigned local variable 'y1'
+                //         y1.ToString();
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y1").WithArguments("y1").WithLocation(13, 9),
+                // (19,9): error CS0165: Use of unassigned local variable 'y2'
+                //         y2.ToString();
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y2").WithArguments("y2").WithLocation(19, 9)
+            );
+        }
```

Comment:
>} [](start = 8, length = 1)

Can you also test the simple un-assigned cases of `x ??= y;` where either `x` or `y` is un-assigned?

Reply:
Where x is unassigned is already tested (`c ??= new C()`), I'll add a line for the rhs.

---
In reply to: [200792496](https://github.com/dotnet/roslyn/pull/28253#discussion_r200792496) [](ancestors = 200792496)